tools/wptrunner/wptrunner/wpttest.py

```diff
@@ -211,19 +211,24 @@ class Test:
     result_cls = None  # type: ClassVar[Type[Result]]
     subtest_result_cls = None  # type: ClassVar[Type[SubtestResult]]
     test_type = None  # type: ClassVar[str]
+    pac = None
 
     default_timeout = 10  # seconds
     long_timeout = 60  # seconds
 
     def __init__(self, url_base, tests_root, url, inherit_metadata, test_metadata,
-                 timeout=None, path=None, protocol="http", subdomain=False):
+                 timeout=None, pac=None, path=None, protocol="http", subdomain=False):
         self.url_base = url_base
         self.tests_root = tests_root
         self.url = url
         self._inherit_metadata = inherit_metadata
         self._test_metadata = test_metadata
         self.timeout = timeout if timeout is not None else self.default_timeout
         self.path = path
+        # self.pac will be an absolute path
+        if pac is not None:
+            self.pac = urljoin(urljoin(url_base, os.path.relpath(path, tests_root)), pac)
```

Comment:
Isn't it just resolved relative to `self.url`? Mixing paths and URLs like this seems likely to break on Windows.